designs/2019-changing-default-of-resolve-plugins-relative-to/README.md

```diff
@@ -0,0 +1,94 @@
+1. If `--config` CLI option is present, adopt the directory that contains the config file.
+1. Otherwise, adopt the current working directory.
+
+This setting is used in the whole eslint process. Therefore, every rule ID is identified as unique, so the `rulesMeta` of formatters works fine.
+
+### `resolvePluginsRelativeTo` constructor option of CLIEngine
+
+This RFC changes the default value of `resolvePluginsRelativeTo` option.
+
+1. If `resolvePluginsRelativeTo` constructor option is present, adopt it.
+1. If `configFile` constructor option is present, adopt the directory that contains the config file.
+1. Otherwise, adopt the current working directory.
+
+This setting is used in the whole `CLIEngine` instance. Therefore, every rule ID is identified as unique, so the `CLIEngine#getRules()` method works fine.
+
+### About personal config file (`~/.eslintrc`)
+
+This RFC doesn't change the plugin base path for the personal config file (`~/.eslintrc`) because the functionality has been deprecated in [RFC 32](https://github.com/eslint/rfcs/pull/32). ESLint still loads plugins from the current working directory if there is no project config file, even if there is the personal config file.
+
+If people want to use the personal config file, use `--config` option (e.g., `eslint . --config ~/.eslintrc.json`). It should work intuitively.
+
+### Special handling of `~/`
+
+Bash replaces `~` by the user home directory, but this behavior is not compatible with other platforms. This is similar to that we cannot use glob patterns on every platform.
+
+Therefore, this RFC adds handling of `~`. In the following constructor options, if the path was exactly `~` or started with `~/` (and `~\` as well on Windows), the `CLIEngine` class replaces the `~` by `require("os").homedir()`.
+
+- `cacheLocation`
+- `configFile`
+- `cwd`
+- `ignorePath`
+- `resolvePluginsRelativeTo`
+
+Notes:
+
+- It doesn't handle `~username/` notation because Node.js doesn't provide the way that gets the home directory of other users.
+- It doesn't handle `~` in config files because this is a fallback of stuff that some shells do.
+
+## Documentation
+
+- It needs an entry in the migration guide because of a breaking change.
+- It updates the "[Command Line Interface](https://eslint.org/docs/user-guide/command-line-interface)" page to describe the `--config` option changes the default value of the `--resolve-plugins-relative-to` option.
+- It updates the "[Command Line Interface](https://eslint.org/docs/user-guide/command-line-interface)" page to describe the `--config` option changes the default value of the `--resolve-plugins-relative-to` option.
+- It updates the "[Node.js API](https://eslint.org/docs/developer-guide/nodejs-api)" page to describe the `configFile` constructor option changes the default value of the `resolvePluginsRelativeTo` constructor option on CLIEngine.
+
+## Drawbacks
+
+- The new logic is complex a bit. It increases maintenance cost.
```

Comment:
Add

- Given a personal config file, user might feel encouraged to install packages into the HOME dir. See discussion above.

Probably remove "The new logic is complex a bit" because it is **not**.